src/WCSimPrimaryGeneratorMessenger.cc

```diff
@@ -82,7 +82,16 @@ WCSimPrimaryGeneratorMessenger::WCSimPrimaryGeneratorMessenger(WCSimPrimaryGener
   injectorWavelengthCmd->SetGuidance("Wavelength of the injector laser in nm");
   injectorWavelengthCmd->SetParameterName("injector_wavelength",true);
   injectorWavelengthCmd->SetDefaultValue(435.);
-   
+
+  ibdspectrumCmd = new G4UIcmdWithAString("/mygen/ibd_spectrum",this);
+  ibdspectrumCmd->SetGuidance("Select the IBD spectrum file.");
+  ibdspectrumCmd->SetParameterName("ibd_spectrum", true);
+  ibdspectrumCmd->SetDefaultValue("data/DSNBFluxes.json");
```

Comment:
I don't know what the purpose of `SetDefaultValue()` actually is, but it doesn't do what you think it does. In order to set it, you need this extra call
```suggestion
  ibdspectrumCmd->SetDefaultValue("data/DSNBFluxes.json");
  SetNewValue(ibdspectrumCmd, "data/DSNBFluxes.json");
```

Reply:
Changed the name of this command from `ibdspectrumCmd` to `ibdDatabaseCmd` to better reflect the function of the command. Have also added `SetNewValue` line for this.